src/client/options.rs

```diff
@@ -1455,39 +1446,34 @@ impl ConnectionString {
             host_list.push(address);
         }
 
-        let hosts = if srv {
-            if host_list.len() != 1 {
-                return Err(ErrorKind::InvalidArgument {
-                    message: "exactly one host must be specified with 'mongodb+srv'".into(),
+        let host_info = if !srv {
```

Comment:
This isn't directly related to the change, but when I was reading through to find understand the original code I had a hard time skimming this section.  This changes it to put the success cases up front and have the error cases nicely enumerated.